pkg/apis/wksprovider/machine/os/os.go

```diff
@@ -367,6 +377,58 @@ func (o OS) CreateSeedNodeSetupPlan(params SeedNodeParams) (*plan.Plan, error) {
 	return createPlan(b)
 }
 
+// Sets the IPALLOC_ADDR env var in weave-net based on the servicesCIDRBlocks read from the cluster.yaml
+func setCIDRBlocks(manifests [][]byte, servicesCIDRBlocks []string) ([][]byte, error) {
+	manifestList := &v1.List{}
+	// Parse the manifest from weave-net.yaml into a List
+	err := yaml.Unmarshal(manifests[0], manifestList)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse weave-net daemonset manifest")
+	}
+
+	// Parse the DaemonSet included in the list into an object
+	weaveNetDaemonSet := &appsv1.DaemonSet{}
+	err = yaml.Unmarshal(manifestList.Items[5].Raw, weaveNetDaemonSet)
```

Comment:
It would be better to find the DaemonSet in the list instead of assuming it will be the 6th one.

Reply:
Added a loop to iterate over the manifests listed in weave-net.yaml for this. The daemonset is recognized by it including `containers`, not sure if there is a better way. The manifests are unmarshalled as [`[]runtime.RawExtension`](https://godoc.org/k8s.io/apimachinery/pkg/runtime#RawExtension) and I didn't find the way to get their type